packages/core/useElementHover/index.ts

```diff
@@ -1,13 +1,38 @@
 import type { Ref } from 'vue-demi'
-import { ref } from 'vue-demi'
-import type { MaybeComputedRef } from '@vueuse/shared'
+import { computed, ref } from 'vue-demi'
 import { useEventListener } from '../useEventListener'
+import type { MaybeElementRef } from '../unrefElement'
+import { unrefElement } from '../unrefElement'
+import type { ConfigurableWindow } from '../_configurable'
 
-export function useElementHover(el: MaybeComputedRef<EventTarget | null | undefined>): Ref<boolean> {
+export interface MouseHoverOptions extends ConfigurableWindow {
+  enterDelay?: number
+  leaveDelay?: number
+}
+
+export function useElementHover(el: MaybeElementRef, options: MouseHoverOptions = {}): Ref<boolean> {
+  const enterDelay = options ? options.enterDelay : 0
+  const leaveDelay = options ? options.leaveDelay : 0
   const isHovered = ref(false)
+  const timer = ref<any>(undefined)
```

Comment:
```suggestion
  let timer: typeof ReturnType<setTimeout> | undefined
```

It doesn't need to be ref